src/app/components/AddEvaluationForm/AddEvaluationForm.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useState } from 'react';
+import { Rating } from 'react-simple-star-rating';
+import styled from 'styled-components';
+import useEvaluations from '../../hooks/useEvaluations';
+import type { Pupil } from '../../types/types';
+import Button from '../Button/Button';
+import Navigation from '../Navigation/Navigation';
+
+type AddEvaluationFormProps = {
+  pupil: Pupil;
+  onSubmit: (category: string, evaluation: string) => void;
+  onCancel: () => void;
+  missingInput: boolean;
+};
+
+export default function AddEvaluationForm({
+  pupil,
+  onSubmit,
+  onCancel,
+}: AddEvaluationFormProps): JSX.Element {
+  const { evaluations } = useEvaluations();
+  const [selectedCategory, setSelectedCategory] = useState('');
+  const [selectedEvaluation, setSelectedEvaluation] = useState('');
+  const [rating, setRating] = useState(0);
+  const [inputError, setInputError] = useState(false);
+  console.log(evaluations);
+
+  function handleRating(rate: number) {
+    setRating(rate);
+  }
+
+  function handleCategory(category: string) {
+    setSelectedCategory(category);
+  }
+
+  function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    if (selectedCategory === '' || rating === 0 || selectedEvaluation === '') {
+      setInputError(true);
+      return;
+    } else {
+      const category = selectedCategory;
+      const evaluation = selectedEvaluation;
+      onSubmit(category, evaluation);
+      setInputError(false);
+      setSelectedCategory('');
+      setSelectedEvaluation('');
+    }
+  }
```

Comment:
when you return in your if statement, then you dont have to wrap the rest into an else statement.